cpu/esp_common/periph/uart.c

```diff
@@ -108,23 +108,24 @@ static struct uart_hw_t _uarts[] = {
         .int_src = ETS_UART0_INTR_SOURCE
 #endif /* !MCU_ESP8266 */
     },
-#if defined(UART1_TXD) || !defined(MCU_ESP8266)
+
+#if defined(UART1_TXD) && defined(UART1_RXD) && !defined(MCU_ESP8266)
     {
         .regs = &UART1,
         .used = false,
         .baudrate = STDIO_UART_BAUDRATE,
         .data = UART_DATA_BITS_8,
         .stop = UART_STOP_BITS_1,
         .parity = UART_PARITY_NONE,
-#if !defined(MCU_ESP8266)
         .mod = PERIPH_UART1_MODULE,
         .signal_txd = U1TXD_OUT_IDX,
         .signal_rxd = U1RXD_IN_IDX,
         .int_src = ETS_UART1_INTR_SOURCE
-#endif /* !MCU_ESP8266 */
     },
-#endif /* defined(UART1_TXD) || !defined(MCU_ESP8266) */
-#if defined(CPU_FAM_ESP32)
+#endif /* defined(UART1_TXD) && defined(UART1_RXD) && !defined(MCU_ESP8266) */
+
+#if defined(UART2_TXD) && defined(UART2_RXD) && \
+    (defined(CPU_FAM_ESP32) || defined(CPU_FAM_ESP32S3))
```

Comment:
```suggestion
    defined(ETS_UART2_INTR_SOURCE)
```

Would that work? (or `!defined(MCU_ESP8266)`?)

Reply:
Unfortunatly not since `ETS_UART*_INTR_SOURCE` are only defined for ESP32 :worried: but I have changed it to
```suggestion
    #if defined(UART2_TXD) && defined(UART2_RXD) && (SOC_UART_NUM > 2)
```